scripts/security/tocve.py

```diff
@@ -0,0 +1,225 @@
+#! /usr/bin/env python3
+
+import json
+import re
+import subprocess
+import sys
+
+# TODO: impact and problemtype.
+# TODO: setup publication date.
+
+# Setup Instructions
+# ==================
+#
+# You will need to install the [Atlassian Command Line
+# Interface](https://bobswift.atlassian.net/wiki/spaces/ACLI/overview)
+# Set this variable to where you have installed this tool.
+acli = '/home/<user>/ACLI/acli'
+
+# To use this script, you will need to fill in your username, and you
+# will need to generate an API token.  These can be generated at:
+# https://id.atlassian.com/manage-profile/security/api-tokens
+user = '<email>'
+token = '<token>'
+
+# You can verify this works with the following command:
+# ```
+# ./acli jira -u <email> -t <token> --action getProjectList \
+#       --server https://zephyrprojectsec.atlassian.net
+# ```
+#
+# Note that only members of the triage team will have access to issues
+# that are under embargo.
+#
+# Usage
+# =====
+#
+# Run as either `./tocve.py ZEPSEC-nn cve` to generate a CVE or as
+# `./tocve.py ZEPSEC-nn rnote` to generate release notes.  The CVE
+# command expects a cve-template.json in the current directory, which
+# can be updated with values for the fields this script does not set.
+
+# Set this to where you have installed ACLI.
+acli_base = [acli, 'jira', '-u', user,
+        '-t', token,
+        '--quiet',
+        '--outputType', 'json',
+        '--server', 'https://zephyrprojectsec.atlassian.net' ]
+
+url_base = 'https://zephyrprojectsec.atlassian.net/browse/'
+vul_doc = "https://docs.zephyrproject.org/latest/security/vulnerabilities.html"
+
+def flatten_json(j):
+    """Flatten a json input.  Given an array of single map entries,
+    fold them all into a single object."""
+    obj = {}
+    for item in j:
+        for k, v in item.items():
+            if k in obj:
+                raise Exception("Duplicate key: {}".format(k))
+            obj[k] = v
+    return obj
+
+def query(issue, action, *args):
+    cmd = acli_base.copy()
+    cmd.extend(['--issue', issue, '--action', action])
+    cmd.extend(args)
+    lines = subprocess.check_output(cmd)
+    return json.loads(lines)
+
+def query_ticket(issue):
+    data = query(issue, "getIssue")
+    return data
+
+def query_weblinks(issue):
+    data = query(issue, "getRemoteLinkList")
+    data = [flatten_json(x) for x in data]
+    return data
+
+def query_links(issue):
+    data = query(issue, "getLinkList")
+    data = [flatten_json(x) for x in data]
+    return data
+
+def clean_version(verno):
+    return re.sub(r' \(\d+\)', '', verno)
+
+class Issue():
+    def __init__(self, issue):
+        self.issue = issue
+        # print("Querying {}".format(issue))
+        self.ticket = query_ticket(issue)
+        self.weblinks = query_weblinks(issue)
+        self.links = query_links(issue)
+        # print("ticket: ", self.ticket)
+        # print("links: ", self.links)
+        # print("done")
```

Comment:
don't forget remote these prints for debug

Reply:
Do they hurt anything? Well, at least the ones that can print out the ticket or links. The 'done' not so useful.

Reply:
we don't have these leftovers in other scripts